OPEN_PROBLEMS/PUBSUB_AT_SCALE.md

```diff
@@ -0,0 +1,128 @@
+- GoCast: Gossip-enhanced Overlay Multicast for Fast and Dependable Group Communication, 2005, http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.75.4811&rep=rep1&type=pdf
+- Exposing and eliminating vulnerabilities to denial of service attacks in secure gossip-based multicast, DSN 2004
+- Emergent structure in unstructured epidemic multicast, DSN 2007
+
+
+##### peer-base collaboration messaging protocol (Dias Peer Set)
+
+- https://github.com/peer-base/peer-base/blob/master/docs/PROTOCOL.md#peer-base---protocol-explanation
+
+### Within the broad Research Ecosystem
+> How do people try to solve this problem?
+
+- We've collected a vast amount of Research Material around PubSub. It can be found at https://github.com/libp2p/research-pubsub
+
+Related literature in the broader research community has worked towards two incarnations of pubsub systems: i) content-based pubsub and ii) topic-based pubsub.
+
+<strong> Content-Based Pubsub </strong>
+
+In content-based pubsub, newly published content is tagged with a set of attribute/value pairs. In turn, subscriptions are expressed as predicates of the attributes. When predicates match attributes, the subscriber receives the information.
+
+Related literature
+
+- Pietzuch, P., Bacon, J.: Hermes: a distributed event-based middleware architecture. In: ICDCS Workshops. pp. 611–618 (2002)
+- Rosenblum, D.S., Wolf, A.L.: A design framework for internet-scale event observation and notification. In: ESEC. pp. 344–360 (1997)
+- Ahmed, N., Linderman, M., Bryant, J.: Papas: Peer assisted publish and subscribe. In: Workshop on Middleware for Next Generation Internet Computing (MW4NG). pp. 7:1–7:6 (2012)
+- Li, M., Ye, F., Kim, M., Chen, H., Lei, H.: A scalable and elastic publish/subscribe service. In: IPDPS. pp. 1254–1265 (2011)
+- Zhang, B., Jin, B., Chen, H., Qin, Z.: Empirical evaluation of contentbased pub/sub systems over cloud infrastructure. In: Intl. Conference on Embedded and Ubiquitous Computing (EUC). pp. 81–88 (2010)
+
+<strong> Topic-Based Pubsub </strong>
+
+In contrast, in topic-based pubsub systems, subscribers declare interest in some topic. Publishers add/tag topics to their publication and those topics that match subscribers' interests are broadcast to them. Depending on the nature of the application running on top, topic-based pubsub can reduce the complexity of the system.
+
+Related Literature
+
+- Eugster, P.T., Felber, P.A., Guerraoui, R., Kermarrec, A.M.: The many faces of publish/subscribe. ACM Comput. Surv. 35(2), 114–131 (2003)
+- Zhao, Y., Kim, K., Venkatasubramanian, N.: Dynatops: A dynamic topicbased publish/subscribe architecture. In: DEBS. pp. 75–86 (2013)
+- Julien Gascon-Samson, Franz-Philippe Garcia, Bettina Kemme, Jörg Kienzle, Dynamoth: A Scalable Pub/Sub Middleware for Latency-Constrained Applications in the Cloud, IEEE ICDCS 2015
+
+
+### Known shortcommins of existing solutions
+> What are the limitations on those solutions?
+
```

Comment:
The known shortcomings is currently empty.

Reply:
One that should be highlighted is that almost no PubSub literature covers Tree Forming, they all mostly assume that the number of peers is consistent (no churn) and that they can dictate the spanning tree to their own design at any moment in time (without contemplating the challenges with having an accurate perspective of the network).

It would be valuable to highlight some papers that focus on efficient routing at the underlay level (i.e. OSPF, RIP, etc) to contrast and take inspiration from.